src/driver-etekcity.c

```diff
@@ -457,6 +457,9 @@ etekcity_read_profile(struct ratbag_profile *profile)
 	}
 
 	ratbag_profile_for_each_resolution(profile, resolution) {
+		if (!resolution->dirty)
+			continue;
+
```

Comment:
you are in `etekcity_read_profile()`, you probably shouldn't be checking for the dirty flag.

Also, etekcity and roccat are both using the old write profile stubs, and so they do not need to check for the dirty flag as you are already checking it in `libratbag.c`